neps/nep-0366.md

```diff
@@ -0,0 +1,62 @@
+
+## Motivation
+
+NEAR has been designed with simplify of onboarding in mind. One of the large hurdles right now is that after creating an implicit or even named account user doesn't have NEAR to pay gas fees to interact with apps.
+
+For example, apps that pay user for doing work (like NEARCrowd or Sweatcoin) or free-to-play games.
+
+[Aurora Plus](https://aurora.plus) has shown viability of the relayers that can offer some number of free transactions and a subscription model. Shifting the complexity of dealing with fees to the infrastructure from the user space.
+
+## Rationale and alternatives
+
+Proposed here design provides the easiest for users and developers way to onboard and pay for user transactions.
+There is no requirement on the user account, including user account may not even exist on chain and implicit account can be used.
+
+An alterantive is a proxy contracts deployed on the user account. 
+This design has severe limitations as it requires user to deploy such contract and additional costs for storage.
+
+## Specification
+
+Next changes to protocol are required:
+ - New Action kind: `DelegateAction(receiver_id, Action, signer_pk, signature)`
+ - On conversion to Receipt, such action is sent as `DelegateAction` to the `receiver_id`.
+ - On processing Receipt, it verifies the `signature` over the given account and `signer_pk`. Unwraps into `receiver_id: AccountId`, `action: Action` inside into new Receipt with given `receiver_id` and `action`. This means that `predeccessor_id` of such action has been overriden.
+
+See the [DelegateAction specification](specs/RuntimeSepc/Actions.md#DelegateAction) for details.
+
+## Security Implications
+
+Delegate actions do not override `signer_pk`, leaving that to the original signer that initiated transaction (eg relayer in meta transaction case). Although it is possible to override `signer_pk` in the context with one from `DelegateAction`, there is no clear value to do that.
+
+See ***Validation*** section in [DelegateAction specification](specs/RuntimeSepc/Actions.md#DelegateAction) for security considerations around what user signs and validation of actions with different permissions.
+
+## Drawbacks
+
+Increases complexity of the NEAR's transactional model.
+
+Meta transactions take an extra block to execute, as they first need to be included by the originating account, then routed to the delegate account and only after that to the real destination.
+
+Delegate actions are not programmable as they require having signatures. Original proposal contained a new `AccessKey` kind that would support programmable delegated actions. On the other hand, that would be limiting without progamability of smart contracts, hence that idea evolved into [Account Extensions](https://github.com/nearprotocol/neps/pull/0000).
+
+## Future possibilities
+
+Supporting ZK proofs instead of just signatures can allow for anonymous transactions, which pay fees to relayers in anonymous way.
```

Comment:
```suggestion
Supporting ZK proofs instead of just signatures can allow for anonymous transactions, which pay fees to relayers in an anonymous way.
```

Reply:
It's better to say "in a confidential way". If we are talking about anonymity, then ring signatures could be applied. Please excuse (and correct me) if I misunderstood and the above does not make sense.

Reply:
The difference between two is not that big in case of ZK based pools.
External observer can not tell who is transacting with whom. As a receiver or sender you also do not know how much the other party has transacted and owns. Due to that you are able to send and receive payments from anonymous parties.